apps/settings/js/panels/call_forwarding/panel.js

```diff
@@ -0,0 +1,248 @@
+define(function(require) {
+  'use strict';
+
+  var CallConstant = require('panels/call/call_constant');
+  var CallForwarding = require('panels/call/call_forwarding');
+  var CallUtils = require('panels/call/call_utils');
+  var DialogService = require('modules/dialog_service');
+  var SettingsCache = require('modules/settings_cache');
+  var SettingsPanel = require('modules/settings_panel');
+  var mozSettings = navigator.mozSettings;
+
+  return function ctor_call_barring() {
+    return SettingsPanel({
+      onInit: function(panel, options) {
+        this._elements = {};
+        this._elements.panel = panel;
+
+        this._cardIndex = options.cardIndex || 0;
+        this._conns = window.navigator.mozMobileConnections;
+        this._conn = this._conns[this._cardIndex];
+
+        this._elements.liCfuDesc = panel.querySelector('#li-cfu-desc');
+        this._elements.liCfmbDesc = panel.querySelector('#li-cfmb-desc');
+        this._elements.liCfnRepDesc = panel.querySelector('#li-cfnrep-desc');
+        this._elements.liCfnReaDesc = panel.querySelector('#li-cfnrea-desc');
+
+        this._elements.cfuDesc = panel.querySelector('#cfu-desc');
+        this._elements.cfmbDesc = panel.querySelector('#cfmb-desc');
+        this._elements.cfnRepDesc = panel.querySelector('#cfnrep-desc');
+        this._elements.cfnReaDesc = panel.querySelector('#cfnrea-desc');
+
+        this._elements.mobileBusyMenuItem =
+          panel.querySelector('.callForwardingMobileBusy');
+        this._elements.noReplyMenuItem =
+          panel.querySelector('.callForwardingNoReply');
+        this._elements.notReachableMenuItem =
+          panel.querySelector('.callForwardingNotReachable');
+        this._elements.unConditionalMenuItem =
+          panel.querySelector('.callForwardingUnconditional');
+
+        this._bindMenuItemsClickEvent();
+      },
+
+      onBeforeShow: function(panel, options) {
+        if (options.cardIndex) {
+          this._cardIndex = options.cardIndex;
+          this._conn = this._conns[this._cardIndex];
+        }
+
+        this._updateSubpanels();
```

Comment:
nit: This name seems out of dated to me as call forwarding is not a sub panel any more.